apps/trading/components/swap/swap-container.tsx

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  useAssetsMapProvider,
+  type AssetFieldsFragment,
+} from '@vegaprotocol/assets';
+import {
+  getBaseAsset,
+  getQuoteAsset,
+  isSpot,
+  marketDataProvider,
+  useMarketsMapProvider,
+  type MarketFieldsFragment,
+} from '@vegaprotocol/markets';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useAccounts } from '@vegaprotocol/accounts';
+import { useDataProvider } from '@vegaprotocol/data-provider';
+import { useNavigate } from 'react-router-dom';
+import { SwapForm } from './swap-form';
+import { Side } from '@vegaprotocol/types';
+
+export const SwapContainer = () => {
+  const { pubKey } = useVegaWallet();
+  const { data: markets } = useMarketsMapProvider();
+  const { data: assetsData } = useAssetsMapProvider();
+  const { data: accounts } = useAccounts(pubKey);
+
+  const [quoteAsset, setQuoteAsset] = useState<AssetFieldsFragment>();
```

Comment:
I don't understand why there is so much state requried. As far as I can see everything derives off the two selected assets? This could be much more succinct